solr/core/src/java/org/apache/solr/schema/IndexSchema.java

```diff
@@ -190,10 +197,41 @@ public IndexSchema(String name, InputSource is, Version luceneVersion, SolrResou
   protected IndexSchema(Version luceneVersion, SolrResourceLoader loader, Properties substitutableProperties) {
     this.luceneVersion = Objects.requireNonNull(luceneVersion);
     this.loader = loader;
-    this.solrClassLoader = loader;//loader.getCore() == null? loader: loader.getCore().getSchemaPluginsLoader();
+    this.solrClassLoader = createClassLoader(loader);
     this.substitutableProperties = substitutableProperties;
   }
 
+  private SolrClassLoader createClassLoader(SolrResourceLoader loader) {
+    if(loader.getCoreContainer() != null && loader.getSolrConfig() != null){
+      return new PackageListeningClassLoader(loader.getCoreContainer(), loader,
+          pkg -> loader.getSolrConfig().maxPackageVersion(pkg),
+          () -> reloadCoreSchema());
+    }
+    return loader;
+  }
+
+  private void reloadCoreSchema() {
+    if(coreContainer != null) {
+     cores.forEach((coreName, id) -> {
+       try(SolrCore core = coreContainer.getCore(coreName, id)){
+         if(core != null) {
+           core.fetchLatestSchema();
+         }
+       }
+     });
+    }
+  }
```

Comment:
Each IndexSchema object gets a `PackageListeningClassLoader` with a listener that, on change, triggers a reload of the schema in all cores?

Reply:
This happens only if you share the schema. If schema is not shared it only results in reloading schema in that one core